toolchain/lc-compile/src/report.c

```diff
@@ -152,6 +169,22 @@ static void _PrintPosition(long p_position)
     fprintf(stderr, "%s:%ld:%ld: ", t_path, t_row, t_column);
 }
 
+/* Print the source code line that contains p_position, and another
+ * line below it with a caret pointing "up" to the exact character
+ * that the position specifies. */
+static void _PrintContext(long p_position)
```

Comment:
@montegoulding IMHO it should be the default output format, just like pretty much every other compiler out there.

At the moment we don't have anything that parses the output that could get confused by this.  In the long term, when we actually have a tool that can't we can probably add a designed-to-be-machine-readable output format for messages (for example, @runrevmark has suggested adding a `--json` mode where _all_ output is JSON on stdout).

Note that all "context" output is guaranteed to be on lines starting with whitespace.